Fixed Bug

Musaevt opened the pull request:
Was a problem with timestamp filtering in GET /api/Media call

Musaevt committed:
Fixed Bug
Was a problem with timestamp filtering in GET /api/Media call

SvenEV commented on HiP-DataStore/Controllers/MediaController.cs:
```diff
@@ -78,16 +79,22 @@ public IActionResult Get(MediaQueryArgs args)
-                    .Select(x => new MediaResult
+                    .ToList();
+
+                medias = medias.AsQueryable().FilterIf(args.Timestamp != null, x => DateTimeOffset.Compare(x.Timestamp, args.Timestamp.GetValueOrDefault()) == 1).ToList();
```
Hm, I think filtering after pagination is dangerous... Imagine at line 85 there are 10 items and a page size of 5, then line 89 would return the first 5 items (i.e. the first page). Now, filtering by timestamp is done AFTER pagination, so line 92 might throw out some of these 5 items. Thus, in the end you might end up with page 1 of 2 having less items than the desired page size.

SvenEV replied:
I guess your problem was that MongoDB does not support DateTimeOffset.Compare(...) in queries, right? Hm, I'll have a look at that issue and maybe find a workaround.

Musaevt committed:
Merge branch 'develop' into HIPCMS-632

# Conflicts:
#	HiP-DataStore/Controllers/MediaController.cs

Musaevt committed:
Removed direct access to Read Model in PutFileById method

SvenEV commented on HiP-DataStore/Controllers/MediaController.cs:
```diff
@@ -209,27 +211,23 @@ public IActionResult GetFileById(int id)
             // Remove old file
-            if (media.File != null && System.IO.File.Exists(media.File))
-                System.IO.File.Delete(media.File);
+            string FilePath = _mediaIndex.GetFilePath(id);
```
"FilePath" should be named "filePath"

SvenEV commented on HiP-DataStore/Core/WriteModel/MediaIndex.cs:
```diff
@@ -25,6 +26,24 @@ public bool IsPublishedAudio(int id)
         }
+        public MediaType? GetMediaType(int id)
+        {
+            lock (_lockObject)
```
Locking is a good idea, but it only works if all calls to the _media dictionary use locking

Musaevt committed:
-- Changed style of variable definition
-- Added locks to MediaIndex
-- Added handling events MediaUpdate, FileUpdate for DB

SvenEV reviewed (approved):
Looks good!

Musaevt committed:
Deleted redundant property name
